misk-hibernate/src/main/kotlin/misk/hibernate/HibernateEntityModule.kt

```diff
@@ -20,19 +25,66 @@ abstract class HibernateEntityModule(
 
   override fun configure() {
     // Initialize empty sets for our multibindings.
+    newMultibinder<DatabaseQueryMetadata>()
     newMultibinder<HibernateEntity>(qualifier)
+    newMultibinder<HibernateQuery>()
     newMultibinder<ListenerRegistration>(qualifier)
 
     configureHibernate()
   }
 
+  protected fun installHibernateAdminDashboardWebActions() {
+    install(HibernateDatabaseQueryWebActionModule())
+  }
+
   protected fun addEntities(vararg entities: KClass<out DbEntity<*>>) {
-    for (entity in entities) {
-      multibind<HibernateEntity>(qualifier)
-          .toInstance(HibernateEntity(entity))
+    entities.forEach { entity ->
+      addEntity(dbEntityClass = entity)
+    }
+  }
+
+  /** Install entities with an AccessAnnotation to set run query permissions for Database-Query tab */
+  protected fun addEntities(accessAnnotationClass: KClass<out Annotation>, vararg entities: KClass<out DbEntity<*>>) {
+    entities.forEach { entity ->
+      addEntity(dbEntityClass = entity, accessAnnotationClass = accessAnnotationClass)
     }
   }
 
+  /** Install Entity with a default of no query access from Admin Dashboard */
+  protected fun <T : DbEntity<T>> addEntity(
+    dbEntityClass: KClass<T>,
+    queryClass: KClass<out Query<T>>? = null,
+    accessAnnotationClass: KClass<out Annotation> = NoAdminDashboardDatabaseQueryAccess::class
+  ) {
+    multibind<HibernateEntity>(qualifier)
+        .toInstance(HibernateEntity(dbEntityClass))
+
+    multibind<DatabaseQueryMetadata>().toProvider(DatabaseQueryMetadataProvider(
+        dbEntityClass = dbEntityClass,
+        queryClass = queryClass,
+        accessAnnotationClass = accessAnnotationClass
+    ))
+    if (queryClass != null) {
+      multibind<HibernateQuery>().toInstance(
+          HibernateQuery(queryClass as KClass<out Query<DbEntity<*>>>))
+    }
+  }
+
+  /** Install Entity with a default of no query access from Admin Dashboard */
+  protected inline fun <reified T: DbEntity<T>>addEntity() {
+    addEntity(T::class)
+  }
+
+  /** Adds a DbEntity to Database-Query Admin Dashboard Tab with a dynamic query (not a static Misk.Query) */
+  protected inline fun <reified T : DbEntity<T>, reified AA: Annotation> addEntityWithDynamicQuery() {
+    addEntity(T::class, null, AA::class)
+  }
+
+  /** Adds a DbEntity to Database-Query Admin Dashboard Tab with a static Misk.Query */
```

Comment:
I think it might be structurally simpler to add entities and independently?

Reply:
I originally did it like that but thought it would provide better discoverability of the tab to automatically bind installed entities for the metadata so that the tab is partly functional when folks first go to it in their service.